docs/src/content/en/docs/auth/supabase.mdx

```diff
@@ -0,0 +1,131 @@
+---
+title: "MastraAuthSupabase"
+description: "Documentation for the MastraAuthSupabase class, which authenticates Mastra applications using Supabase Auth."
+---
+
+import { Tabs, Tab } from "@/components/tabs";
+
+# MastraAuthSupabase
+
+The `MastraAuthSupabase` class provides authentication for Mastra using Supabase Auth. It verifies incoming requests using Supabase's authentication system and integrates with the Mastra server using the `experimental_auth` option.
+
+## Installation
+
+Before you can use the `MastraAuthSupabase` class you have to install the `@mastra/auth-supabase` package.
+
+```bash copy
+npm install @mastra/auth-supabase@latest
+```
+
+## Usage example
+
+```typescript {2,7-9} filename="src/mastra/index.ts" showLineNumbers copy
+import { Mastra } from "@mastra/core/mastra";
+import { MastraAuthSupabase } from '@mastra/auth-supabase';
+
+export const mastra = new Mastra({
+  // ..
+  server: {
+    experimental_auth: new MastraAuthSupabase({
+      url: process.env.SUPABASE_URL,
+      anonKey: process.env.SUPABASE_ANON_KEY
+    }),
+  },
+});
+```
+
+> See the [MastraAuthSupabase](/reference/auth/supabase.mdx) API reference for all available configuration options.
+
+## Client-side Setup
+
+When using Supabase auth, you'll need to retrieve the access token from Supabase on the client side and pass it to your Mastra requests.
+
+### Retrieving the access token
+
+Use the Supabase client to authenticate users and retrieve their access token:
+
+```typescript filename="lib/auth.ts" showLineNumbers copy
+import { createClient } from "@supabase/supabase-js";
+
+const supabase = createClient("<supabase-url>", "<supabase-key>");
+
+const authTokenResponse = await supabase.auth.signInWithPassword({
+  email: "<user's email>",
+  password: "<user's password>",
+});
+
+// Retrieve the access token
```

Comment:
Remove comment in the code block, it's kind of explained in the explainer paragraph.